spyder/api/shellconnect/mixins.py

```diff
@@ -92,89 +201,133 @@ class ShellConnectPluginMixin(ShellConnectMixin):
     # ---- Connection to the IPython console
     # -------------------------------------------------------------------------
     @on_plugin_available(plugin=Plugins.IPythonConsole)
-    def on_ipython_console_available(self):
-        """Connect to the IPython console."""
+    def on_ipython_console_available(self) -> None:
+        """
+        Connect this object to the signals of the IPython Console.
+
+        Returns
+        -------
+        None
+        """
         ipyconsole = self.get_plugin(Plugins.IPythonConsole)
         self.register_ipythonconsole(ipyconsole)
 
     @on_plugin_teardown(plugin=Plugins.IPythonConsole)
-    def on_ipython_console_teardown(self):
-        """Disconnect from the IPython console."""
+    def on_ipython_console_teardown(self) -> None:
+        """
+        Disconnect this object from the signals of the IPython Console.
+
+        Returns
+        -------
+        None
+        """
         ipyconsole = self.get_plugin(Plugins.IPythonConsole)
         self.unregister_ipythonconsole(ipyconsole)
 
     # ---- Public API
     # -------------------------------------------------------------------------
-    def set_shellwidget(self, shellwidget):
+    def set_shellwidget(
+        self, shellwidget: spyder.plugins.ipyconsole.widgets.shell.ShellWidget
+    ) -> None:
         """
-        Update the current shellwidget.
+        Set the given shell widget as the one associated to the current widget.
```

Comment:
```suggestion
        Set the stacked widget associated to the given shell widget.
```

Same as my suggestion for `ShellConnectMainWidget`.